src/xref2/cpath.ml

```diff
@@ -186,16 +186,17 @@ and is_resolved_module_hidden :
  fun ~weak_canonical_test ->
```

Comment:
Unrelated to this PR but `weak_canonical_test` is very rarely passed through, almost always set to a constant. Would it make sense to remove this argument but do extra check at the call site ? eg. change `is_resolved_module_hidden ~weak_canonical_test:true p` into `is_weak_canonical p ||  is_resolved_module_hidden p`,

Reply:
I think I prefer it as one function - though we could make the `weak_canonical_test` an optional parameter.